x/amm/keeper/keeper_swap_exact_amount_in.go

```diff
@@ -34,12 +34,12 @@ func (k Keeper) SwapExactAmountIn(
 	}
 	tokensIn := sdk.Coins{tokenIn}
 
-	defer func() {
-		if r := recover(); r != nil {
-			tokenOutAmount = math.Int{}
-			err = fmt.Errorf("function swapExactAmountIn failed due to internal reason: %v", r)
-		}
-	}()
+	// defer func() {
+	// 	if r := recover(); r != nil {
+	// 		tokenOutAmount = math.Int{}
+	// 		err = fmt.Errorf("function swapExactAmountIn failed due to internal reason: %v", r)
+	// 	}
+	// }()
```

Comment:
Why this panic recovery is being removed? Isn't it one derived from Osmosis?

Reply:
Yeah I should recover it. Thanks.

Reply:
only for debugging purposes, we should restore that